### PR TITLE
add profiles to search history

### PR DESCRIPTION
**Pull request by skiniks**
This PR satisfies #2513, a duplicate of #1677. The SearchHistory component is enhanced by fully implementing the display of recent profiles. Each recent profile now includes the avatar, display name, and handle. Additionally, the display name and handle are truncated with ellipses if they exceed a specified character limit to ensure they fit within the given space.

Screenshot 2024-05-22 at 3 32 30 PM

### CODE REVIEW

File: src/view/screens/Search/Search.tsx [hunk 1]
```diff
@@ -562,22 +572,68 @@ export function SearchScreen(
...
+
   const navigateToItem = React.useCallback(
     (item: string) => {
       scrollToTopWeb()
       setShowAutocomplete(false)
       updateSearchHistory(item)
 
       if (isWeb) {
-        navigation.push('Search', {q: item})
+        navigation.replace('Search', {q: item})
```
Comment (gaearon) [inline src/view/screens/Search/Search.tsx:606]: why replace? i think push was intentional

File: src/view/screens/Search/Search.tsx [hunk 2]
```diff
@@ -880,12 +982,66 @@ function SearchHistory({
...
+                      accessibilityIgnoresInvertColors
+                    />
+                    <Text style={[pal.text, styles.profileName]}>
+                      {truncateText(profile.displayName || '', 12)}
+                    </Text>
+                    <Text style={[pal.textLight, styles.profileHandle]}>
+                      @{truncateText(profile.handle, 12)}
+                    </Text>
+                  </Pressable>
+                  <Pressable
```
Comment (gaearon) [inline src/view/screens/Search/Search.tsx:1026]: Does this need accessibility label?

File: src/view/screens/Search/Search.tsx (hunk 1 above)
Comment (skiniks) [reply src/view/screens/Search/Search.tsx]: my mistake! i've reverted that.

File: src/view/screens/Search/Search.tsx (hunk 2 above)
Comment (skiniks) [reply src/view/screens/Search/Search.tsx:1026]: definitely! i've added the label and hint.

**Comment by gaearon** [comment]
A few issues.

Currently, pressing the avatar hides the dropdown before pushing the next view. That causes flicker:

https://github.com/bluesky-social/social-app/assets/810438/e4f8711e-d52f-4fbb-aa1e-a18728d6828c

I think we'd want to avoid that.

Another problem is that pushing an autocomplete result now seems to navigate _twice_.

https://github.com/bluesky-social/social-app/assets/810438/ca43166c-3877-4de5-a41a-1565d4d6af3b

It seems like you're adding some navigation logic — maybe existing logic could be reused? Whatever we're doing for the existing autocomplete. I presume it wouldn't have those issues? But I haven't looked closely.

**Review by gaearon — CHANGES_REQUESTED** [review]
need to fix flicker and double nav

**Comment by skiniks** [comment]
it looks like removing this fixes the issues:

```javascript
scrollToTopWeb()
setShowAutocomplete(false)
updateSelectedProfiles(profile)

**Comment by gaearon** [comment]
seems like Back button is broken


https://github.com/bluesky-social/social-app/assets/810438/df5df32c-4f87-48fd-89fc-f0708712c44e

**Review by gaearon — APPROVED** [review]
thanks!

**Pull request merged**